adam-core/src/main/scala/org/bdgenomics/adam/rdd/ShuffleRegionJoin.scala

```diff
@@ -116,9 +120,11 @@ sealed trait ShuffleRegionJoin[T, U, RT, RU] extends RegionJoin[T, U, RT, RU] {
     if (leftIter.isEmpty || rightIter.isEmpty) {
       emptyFn(leftIter, rightIter)
     } else {
-      val bufferedLeft = leftIter.buffered
-      val currentBin = bufferedLeft.head._1._2
-      val region = bins.value.invert(currentBin)
+      val listRepresentationLeft = leftIter.toList
```

Comment:
I'm aware of that requirement, but that's why we were buffering the left iterator in the original code.